examples/wait_for_scan_results.py

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env python
+
+'''
+Created on May 1, 2019
+
+@author: gsnyder
+
+Wait for scannning results, i.e. after uploading a scan, wait for all the jobs that
+process the scan results to complete
+'''
+
+import argparse
+from datetime import datetime
+import json
+import logging
+import sys
+import time
+import timestring
+
+from blackduck.HubRestApi import HubInstance, object_id
+
+parser = argparse.ArgumentParser("Wait for scan processing to complete for a given code (scan) location/name")
+parser.add_argument("scan_location_name", help="The scan location name")
+parser.add_argument('-m', '--max_checks', type=int, default=10, help="Set the maximum number of checks before quitting")
+parser.add_argument('-t', '--time_between_checks', default=5, help="Set the number of seconds to wait in-between checks")
+parser.add_argument('-s', '--snippet_scan', action='store_true', help="Select this option if you want to wait for a snippet scan to complete along with it's corresponding component scan.")
+args = parser.parse_args()
+
+logging.basicConfig(format='%(asctime)s:%(levelname)s:%(message)s', stream=sys.stderr, level=logging.DEBUG)
+logging.getLogger("requests").setLevel(logging.WARNING)
+logging.getLogger("urllib3").setLevel(logging.WARNING)
+
+hub = HubInstance()
+
+def get_project_and_version_name(project_version_url):
+    # Get the project and version name and return them concatenated for use to lookup relevant jobs
+    response = hub.execute_get(project_version_url)
+    if response.status_code == 200:
+        project_version_obj = response.json()
+    else:
+        raise Exception("Unable to retrieve project version at URL {}".format(
+            project_version_url))
+
+    version_name = project_version_obj['versionName']
+    project_url = hub.get_link(project_version_obj, "project")
+    response = hub.execute_get(project_url)
+    if response.status_code == 200:
```

Comment:
is it possible to have project_version_url return 200 and not the project_url?

Reply:
Well, it shouldn't be possible. I think you're alluding to the assumption that you can only have a version if there is a project that owns it? And I think your point is I can simplify this code by eliminating some of the checks?

Reply:
yes, just wasn't sure if this was something you'd run into that I hadn't yet :)

Reply:
Call it defensive coding style. Sometimes I use assertions in code to denote the _thing that should never happen_ (but sometimes does anyhow). Other times I write, potentially redundant, code to check things. Sometimes I just write bad code - refactoring was invented for people like me! :-)